app/javascript/lib/validation/validate-file-content.js

```diff
@@ -232,12 +237,87 @@ export async function parseMetadataFile(chunker, mimeType, fileOptions) {
       issues = issues.concat(validateUniqueCellNamesWithinFile(line, isLastLine, dataObj))
       issues = issues.concat(validateMetadataLabelMatches(headers, line, isLastLine, dataObj))
       issues = issues.concat(validateGroupColumnCounts(headers, line, isLastLine, dataObj))
+      if (fileOptions.use_metadata_convention) {
+        issues = issues.concat(validateConventionTerms(headers, line, ontologies, knownErrors))
+      }
     // add other line-by-line validations here
     }
   })
   return { issues, delimiter, numColumns: headers[0].length }
 }
 
+/** validate all ontology-based convention terms in a given line */
+export function validateConventionTerms(headers, line, ontologies, knownErrors) {
+  let issues = []
+  const metadataHeaders = headers[0]
+  for (let i = 0; i < metadataHeaders.length; i++) {
+    const header = metadataHeaders[i]
+    if (ONTOLOGY_PROPS.includes(header)) {
+      const ontologyId = line[i]
+      const labelHeader = `${header}${getLabelSuffixForOntology(ontologyId)}`
+      const labelIdx = metadataHeaders.indexOf(labelHeader)
+      const label = line[labelIdx]
+      issues = issues.concat(validateOntologyTerm(header, ontologyId, label, ontologies, knownErrors))
+    }
+  }
+  return issues
+}
+
+/** validate a single ontology ID against stored ontologies and return issues */
+export function validateOntologyTerm(prop, ontologyId, label, ontologies, knownErrors) {
+  const issues = []
+  const ontologyShortNameLc = getOntologyShortNameLc(ontologyId)
+  const ontology = ontologies[ontologyShortNameLc]
+
+  if (ontologyId.includes(':')) {
+    // Convert colon to underscore for ontology lookup
+    const idParts = ontologyId.split(':')
+    ontologyId = `${idParts[0]}_${idParts[1]}`
+  }
+
+  let errorIdentifier
+  let issue
+
+  if (!ontology) {
+    errorIdentifier = `${ontologyId}-label-lookup-error`
+    const accepted = Object.keys(ontologies).join(', ')
+    const msg =
+      `Ontology ID "${ontologyId}" ` +
+      `is not among accepted ontologies (${accepted}) ` +
+      `for key "${prop}"`
+
+    issue = ['error', 'ontology:label-lookup-error', msg]
+  } else if (!(ontologyId in ontology)) {
+    // Register invalid ontology ID
+    const msg = `Invalid ontology ID: ${ontologyId}`
+    errorIdentifier = `${ontologyId}-invalid-id`
+    issue = [
+      'error', 'ontology:label-lookup-error', msg,
+      { subtype: 'ontology:invalid-id' }
+    ]
+  } else {
+    const validLabels = ontology[ontologyId]
+
+    if (!(validLabels.includes(label))) {
+      errorIdentifier = `${ontologyId}-label-lookup-error`
+      // Register invalid ontology label
+      const prettyLabels = validLabels.join(', ')
+      const validLabelsClause = `Valid labels for ${ontologyId}: ${prettyLabels}`
+      const msg = `Invalid ${prop} label "${label}".  ${validLabelsClause}`
+      issue = [
+        'error', 'ontology:label-lookup-error', msg,
```

Comment:
Refining this error type seems [consistent with validation in Ingest Pipeline](https://github.com/broadinstitute/scp-ingest-pipeline/blob/8de8eca097fc3a9e37868142acb57ccb91cbe944/ingest/validation/validate_metadata.py#L1185).

```suggestion
        'error', 'ontology:label-not-match-id', msg,
```